### ISSUE TITLE
:bug: Can no longer access the options menu while media is playing

### ISSUE THREAD

**Issue by jbelew** [jurialmunkey/skin.arctic.fuse.2#270]
### Skin section

Other

### Current Behavior

The title says all. It started with the introduction of the options menu. 

### Expected Behavior

/*

### Steps To Reproduce

/*

### Checklist

- [X] I have made a separate issue for each bug that I am reporting.
- [X] I have searched the issues and this bug has not been reported.
- [X] I am using the most recent skin version.
- [X] I have confirmed that this bug does not occur in the default skin.

**Comment by jurialmunkey** [comment]
Title doesn't say it all 😉

Not surewhat you're referring to.

If you mean pressing back from home opens fullscreen while playing instead of performing the back action, then that is the same behaviour as in Estuary and is hard coded in Kodi

**Comment by jbelew** [comment]
As I mentioned, it worked perfectly before the introduction of the options menu.

Previously, the first backspace would exit to the parent view. From there, subsequent backspaces would take you from the parent, all the way to the widget/home screen view, and pressing backspace again would bring up the left-side menu with access to Settings, etc. Now, the first backspace goes to the parent view, and the second returns you to the currently playing media item. There's no way to progress up the tree. 

I know it was definitely working "as I expect" because I was updating my whitelisting and subtitles settings last week while media was playing.  Additionally, I'd also check the weather while I has music or a radio station was playing, something no longer possible.

**Comment by jbelew** [comment]
> Title doesn't say it all 😉
> 
> Not surewhat you're referring to.
> 
> If you mean pressing back from home opens fullscreen while playing instead of performing the back action, then that is the same behaviour as in Estuary and is hard coded in Kodi

As I mentioned, it worked perfectly before the introduction of the options menu.

Previously, the first Backspace would exit to the parent view. From there, subsequent Backspaces would take you from the parent, all the way up the tree to the widget/home screen view, and pressing Backspace again would bring up the right-side menu with access to Settings, etc. Now, the first Backspace goes to the parent view, and the second returns you to the currently playing media item, the same behavior you'd see using the ESC key. 

I know it was definitely working "as I expect" because I was updating my whitelisting and subtitles settings last week while media was playing. Additionally, I'd also commonly check the weather while I had music or a radio station was playing, something no longer possible.

Addition: I'd dig through the commit history if I had time tonight, but the defect was introduced on or after 10/24. I was last adjusting subtitle settings while watching the final episodes of Vox Machina ;).

**Comment by jurialmunkey** [comment]
There is no way to override this behaviour. It is hardcoded into Kodi.

I'm still not understanding what you mean by "introduction of options menu" -- How can you access the options menu before it existed?

I feel there is some information missing here. My guess... You are using a hub as home. 

The only reason that back "worked" before to open the options menu is because you were never in the Home screen.

The Home screen was faked by forcing a `ReplaceWindow(hubwindowID)` when the Home window loaded. That'd hack it into forcing you back into the hub window whenever you accidentally backtracked too far and opened Home. When a Hub loaded from Home it would then force open the options menu.

This caused all sorts of other problems (no way back to fullscreen, flickering backgrounds, homescreen keymaps ignored etc.) because it was a massive hack.

Now the Home window is correctly used for the Hub when a Hub is set as Home. And *now* you are seeing Kodi's hardcoded behaviour of how the back key works on the Home screen.

This is something that needs to be addressed in Kodi rather than hacked around in a skin. You need to request that Kodi stop overriding the behaviour of the back key when `onback` is defined and not force fullscreen (this behaviour has existed forever).

It is effectively the same issue AppleTV users have where Kodi hardcodes the back key to minimise Kodi on home screen instead of performing the back action.

It's hardcoded behaviour. The skin cannot control it.

**Comment by jurialmunkey** [comment]
Alternatively, you can change your homescreen keymap so that your back remote key does not do `Action(Back)` and instead does `Activatewindow(shutdownmenu)`

I map my longpress back action to open shutdownmenu in my global keymap.

**Comment by L0bsang** [comment]
> I map my longpress back action to open shutdownmenu in my global keymap.

This is a good tip. 
In AF1 I added a full-screen shortcut to the ondown options menu (similar to how stock skin has media controls in the side-blade) so i can get back to the video easily from any menu without having to go all the way to home. I used it a lot.

Seems like in AF2 I can't do that anymore using horizontal views (left wraps, ondown menus are static and don't include media controls), so mapping long-press back to always open options is the way to go! I already do that on homescreen.

FYI your tip about mapping an info button was very nice as well, especially for detailed info on OSD.
